pkg/operator/sync_controllermanager_v311_00.go

```diff
@@ -206,8 +222,108 @@ func manageServiceCatalogControllerManagerDeployment_v311_00_to_latest(client ap
 	case operatorapiv1.Debug:
 		level = 4
 	case operatorapiv1.Normal:
-		level = 3
+		level = 4
+	}
+
+	// ================================================================
+
+	var foundDaemonSet bool
+
+	if proxyConfig != nil {
+		existing, err := client.DaemonSets(required.Namespace).Get(required.Name, metav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			foundDaemonSet = false
+
+			klog.Info("we have a proxyConfig, but no existing daemonset, updating environment")
+			// update the EnvVar
+			addProxyToEnvironment(required, proxyConfig)
+
+			// looks like this is the first time
+			forceRollout = true
+		} else if err != nil {
+			foundDaemonSet = false
+			klog.Error("Problem getting the daemonset, returning")
+			return nil, false, err
+		} else if err == nil {
+			foundDaemonSet = true
+		}
+
+		if foundDaemonSet {
+			// if we have any environments, loop to find the envvars
+			// if we detect a change, force the rollout
+			for _, envVar := range existing.Spec.Template.Spec.Containers[0].Env {
```

Comment:
`required.Spec` will definitely have a single container:
https://github.com/openshift/cluster-svcat-controller-manager-operator/blob/master/bindata/v3.11.0/openshift-svcat-controller-manager/ds.yaml#L24-L25

but `existing` should have one but since it's live data, I'll add some protections against that.

Reply:
So the `Spec.Template.Spec.Containers` must contain at least one pod.

```golang
// List of containers belonging to the pod.
// Containers cannot currently be added or removed.
// There must be at least one container in a Pod.
// Cannot be updated.
// +patchMergeKey=name
// +patchStrategy=merge
Containers []Container `json:"containers" patchStrat...
```

The `DaemonSet` for the Service Catalog has only one container from our definition.

Reply:
I added protections for it anyway in this commit https://github.com/openshift/cluster-svcat-controller-manager-operator/pull/42/commits/14cb711a25cbd9eea46b51e4ec52e88eec778a22